### ISSUE TITLE
Add `encoding` parameter to {read,write}_text (PEP 597).

### ISSUE THREAD

**Issue by liborjelinek** [documatt/sphinx-reredirects#5]
(Migrated from https://gitlab.com/documatt/sphinx-reredirects/-/merge_requests/11. Requested 3 months ago by Anderson Bravalheri @abravalheri.)

With the introduction of PEP 597, relying on the default value of
`encoding` for functions that read/write files is no longer recommended,
and developers have been incentivised to run tests using
`PYTHONWARNDEFAULTENCODING=1` to identify when the `encoding` argument
is not given.

As a result, when `PYTHONWARNDEFAULTENCODING=1` is set on the CI
environment responsible for building the docs, `sphinx_reredirects`
will emit a `EncodingWarning: 'encoding' argument not specified`.

The objective of this change is to comply with the PEP's recommendations
and avoid the `EncodingWarning` from being emitted.

(Proposed changes: https://gitlab.com/documatt/sphinx-reredirects/-/merge_requests/11/diffs?commit_id=93dd8ebe5afcbbd57aa3e70752687b45f319d355.)

**Comment by DimitriPapadopoulos** [comment]
Good idea, this should address warnings that appear in CI jobs such as https://github.com/pypa/setuptools/actions/runs/9179511894/job/25241806696:
```
/home/runner/work/setuptools/setuptools/.tox/docs/lib/python3.12/site-packages/sphinx_reredirects/__init__.py:151: EncodingWarning: 'encoding' argument not specified
  at_path.write_text(content)
```

**Comment by DimitriPapadopoulos** [comment]
For what it's worth, there's a Pylint/ruff rule for that: [PLW1514](https://docs.astral.sh/ruff/rules/unspecified-encoding/)

**Comment by jaraco** [comment]
I encountered this issue today (also in the setuptools project). Anderson's merge request looks right to me. What can we do to get it merged?

**Comment by liborjelinek** [comment]
Fixed in just released v0.1.4.

- https://documatt.com/sphinx-reredirects/rn.html
- https://pypi.org/project/sphinx-reredirects/0.1.4/

Thanks all for patience 😇
